tests/api_connexion/endpoints/test_pool_endpoint.py

```diff
@@ -16,51 +16,143 @@
 import unittest
 
-import pytest
-
+from airflow.models.pool import Pool
+from airflow.utils.session import provide_session
 from airflow.www import app
+from tests.test_utils.db import clear_db_pools
 
 
-class TestPoolEndpoint(unittest.TestCase):
+class TestBasePoolEndpoints(unittest.TestCase):
     @classmethod
     def setUpClass(cls) -> None:
         super().setUpClass()
         cls.app = app.create_app(testing=True)  # type:ignore
 
     def setUp(self) -> None:
         self.client = self.app.test_client()  # type:ignore
+        super().setUp()
+        clear_db_pools()
 
-
-class TestDeletePool(TestPoolEndpoint):
-    @pytest.mark.skip(reason="Not implemented yet")
-    def test_should_response_200(self):
-        response = self.client.delete("/api/v1/pools/TEST_POOL_NAME")
-        assert response.status_code == 204
+    def tearDown(self) -> None:
+        clear_db_pools()
 
 
-class TestGetPool(TestPoolEndpoint):
-    @pytest.mark.skip(reason="Not implemented yet")
-    def test_should_response_200(self):
-        response = self.client.get("/api/v1/pools/TEST_POOL_NAME")
+class TestGetPools(TestBasePoolEndpoints):
+    @provide_session
+    def test_response_200(self, session):
+        pool_model = Pool(pool="test_pool_a", slots=3)
+        session.add(pool_model)
+        session.commit()
+        result = session.query(Pool).all()
+        assert len(result) == 2  # accounts for the default pool as well
+        response = self.client.get("/api/v1/pools")
         assert response.status_code == 200
+        self.assertEqual(
+            {"pools": [{"name": "default_pool",
+                        "slots": 128,
+                        "occupied_slots": 0,
+                        "running_slots": 0,
+                        "queued_slots": 0,
+                        "open_slots": 128,
+                        },
+                       {"name": "test_pool_a",
+                        "slots": 3,
+                        "occupied_slots": 0,
+                        "running_slots": 0,
+                        "queued_slots": 0,
+                        "open_slots": 3,
+                        },
+                       ],
+                "total_entries": 2,
+             },
+            response.json,
+        )
 
-
-class TestGetPools(TestPoolEndpoint):
-    @pytest.mark.skip(reason="Not implemented yet")
-    def test_should_response_200(self):
-        response = self.client.get("/api/v1/pools")
+    @provide_session
+    def test_limit(self, session):
+        pools = [Pool(pool=f"test_pool{i}", slots=1) for i in range(120)]
+        session.add_all(pools)
+        session.commit()
+        result = session.query(Pool).count()
+        self.assertEqual(result, 121)  # accounts for default pool as well
+        # Test for limit under 100
+        response = self.client.get("/api/v1/pools?limit=10")
+        assert response.status_code == 200
+        self.assertEqual(response.json.get("total_entries"), 10)
```

Comment:
```suggestion
        self.assertEqual(response.json["total_entries"], 121)
```
It should return count of all items.

Reply:
Here is the change ce9b79f